Assets/Scripts/UI/TooltipDisplay.cs

```diff
@@ -0,0 +1,56 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+public class TooltipDisplay : MonoBehaviour
+{
+    [Tooltip("The text we want to show")]
+    [SerializeField] public string myString;
+
+    [Tooltip("Reference to tooltip UI element")]
+    [SerializeField] public Canvas tooltip;
+
+    [Tooltip("Reference to Text UI element")]
+    [SerializeField] public Text myText;
+
+    [Tooltip("Determine if we want to display the UI or not")]
+    [SerializeField] public bool displayInfo;
+
+    private void Start()
+    {
+        myText = gameObject.GetComponentInChildren<Text>();
+        tooltip = gameObject.GetComponentInChildren<Canvas>();
```

Comment:
GetComponentInChildren is not really performant. Since the fields for this are public, it would be better if you would just reference them in the inspector

Reply:
For now i will remove the tooltip script and stuffs. Will make a new PR for that.